### PR TITLE
Implementation description for nucDirectory

### PR DESCRIPTION
**Pull request by mgjarrett**
## What is the change?

Adding longer implementation descriptions to `impl` tags in `nucDirectory`.

## Why is the change being made?

This is part of an ongoing SQA effort.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Review by zachmprince — CHANGES_REQUESTED** [review]

File: armi/nucDirectory/nuclideBases.py [hunk 1]
```diff
@@ -12,14 +12,48 @@
...
+    isotopes.
+
+    The :py:class:`NuclideBase <armi.nucDirectory.nuclideBases.NuclideBase>`
+    provides a data structure for information about a single nuclide, including
+    the atom number, atomic weight, element, isomeric state, half-life, and
+    name. 
+
+    The :py:mod:`nuclideBases <armi.nucDirectory.nuclideBases>` module provides
+    a factory and associated functions for instantiating the
+    :py:class:`NuclideBases <armi.nucDirectory.nuclideBases.NuclideBase>` and
```
[the remark this answers is not in the document]
Comment (mgjarrett) [reply armi/nucDirectory/nuclideBases.py:43]: I was trying to make it plural so it feels gramatically correct. Not sure the best way to handle that. Maybe put the `s` outside of the link?

File: armi/nucDirectory/nuclideBases.py [hunk 2]
```diff
@@ -1164,6 +1227,14 @@ def addNuclideBases():
...
         :id: I_ARMI_ND_DATA0
         :implements: R_ARMI_ND_DATA
+
+        This function reads the nuclides.dat file from the ARMI resources
+        folder. This file contains metadata for 4614 nuclides, including
+        number of protons, number of neutrons, atomic number, excited
+        state, element symbol, atomic mass, natural abundance, half-life,
+        and spontaneous fission yield. The data in this file have been
+        collected from multiple different sources; the references are given
+        in comments at the top of the file.
```
[the remark this answers is not in the document]
Comment (mgjarrett) [reply armi/nucDirectory/nuclideBases.py:1238]: I meant at the top of `nuclides.dat`. Maybe I should put that in explicitly instead of saying "the file"

**Comment by john-science** [comment]
`black` linting is broke in this PR.

Sorry, it's a fork repo. I'd fix it for you, but I can't.

File: armi/nucDirectory/nuclideBases.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (john-science) [reply armi/nucDirectory/nuclideBases.py]: Putting an "s" outside the link is what I tried at first (about a month ago).  Sphinx doesn't render it right; it puts a long space between `NuclideBase` and the `s`, like:

> NuclideBase  s

So, I don't think that's the solution.

I think this works fine:

> :py:class:`NuclideBases <...>`

**Comment by mgjarrett** [comment]
@zachmprince I think I've addressed your comments, let me know if you think anything else needs to be changed

**Review by zachmprince — APPROVED** [review]

**Pull request merged**